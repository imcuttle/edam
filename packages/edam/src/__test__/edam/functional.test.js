/* eslint-disable quotes */
/**
 * @file spec
 * @author Cuttle Cong
 * @date 2018/3/28
 * @description
 */
import { mockPrompts } from '../../index'
import { join } from 'path'

describe('functional', function() {
  const tplPath = join(__dirname, '../fixture/edam')
  const outputRoot = join(__dirname, '../fixture/edam-output')

  it('should functional', async () => {
    const ft = await mockPrompts(
      join(tplPath, 'functional'),
      {
        value: 'ojbk'
      },
      join(outputRoot, 'functional')
    )

    expect(Object.keys(ft.tree).length).toBe(3)
    expect(ft.tree).toMatchSnapshot()
  })

  it('should functional output', async function () {
    const ft = await mockPrompts(
      join(tplPath, 'functional'),
      {
        value: 'ojbk'
      },
      join(outputRoot, 'functional')
    )
    expect(await ft.writeToFile()).toBeTruthy()
  })
})
