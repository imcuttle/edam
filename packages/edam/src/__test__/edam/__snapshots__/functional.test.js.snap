// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functional should functional 1`] = `
Object {
  ".gitignore": Object {
    "input": "<%= JSON.stringify(_.file) %>
",
    "loaders": "LoDash",
    "output": "{\\"path\\":\\".gitignore\\",\\"dirname\\":\\".\\",\\"name\\":\\".gitignore\\",\\"ext\\":\\"\\"}
",
  },
  "imignored/keep.module.js": Object {
    "input": "
module.exports = function (variables) {
  return Object.assign({}, {
    name: variables.value,
    version: '2.3.5'
  })
}
",
    "loaders": "module",
    "output": "{
    \\"name\\": \\"ojbk\\",
    \\"version\\": \\"2.3.5\\"
}",
  },
  "index.js": Object {
    "input": "/**
 * @file index
 * @author <%= JSON.stringify(_.file) %>
 * @date 2018/3/28
 * @description
 */

module.exports = '\${value}'
",
    "loaders": "LoDash",
    "output": "/**
 * @file index
 * @author {\\"path\\":\\"index.js\\",\\"dirname\\":\\".\\",\\"name\\":\\"index\\",\\"ext\\":\\".js\\"}
 * @date 2018/3/28
 * @description
 */

module.exports = 'ojbk'
",
  },
}
`;
